### PR TITLE
framework/audio : Fix tinyalsa pcm_wait logic

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
Currently we check only first buffer index, so apb is always null
  Add extra logic to check whole buffers

### CODE REVIEW

File: apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c [hunk 1]
```diff
@@ -883,7 +883,7 @@ static void utc_audio_pcm_writei_p(void)
 		retry = PREPARE_RETRY_COUNT;
 		TC_ASSERT_GEQ_CLEANUP("pcm_writei", ret, 0, clean_all_data(fd, buffer));
 	}
-
+
```
Comment (jsdosa) [inline apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c]: remove white space

**Comment by seinfra** [comment]
Target : [1c4adf29d8fe275391c43f6d6b31734a40d5d71f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1c4adf29d8fe275391c43f6d6b31734a40d5d71f] - Code Rule Check Result:
framework/src/tinyalsa/tinyalsa.c:1440: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

File: framework/src/tinyalsa/tinyalsa.c [hunk 2]
```diff
@@ -1365,105 +1375,109 @@ int pcm_avail_update(struct pcm *pcm)
...
+		if ((pcm->flags & PCM_IN) && !pcm->running && !pcm->draining) {
+			if (pcm_start(pcm) < 0) {
+				return oops(pcm, -1, "Failed to start PCM\n");
+			}
+		}
+	
+		/* Check if some buffer is already available */
+		/* This is put here to handle corner case of open for write -> wait() */
+		if (pcm_avail_update(pcm) == pcm->buffer_size) {
+			auddbg("avail update %d buffer_size %d\n", pcm_avail_update(pcm), pcm->buffer_size);
```
Comment (chanijjani) [inline framework/src/tinyalsa/tinyalsa.c]: audvdbg..?

File: framework/src/tinyalsa/tinyalsa.c [hunk 3]
```diff
@@ -1365,105 +1375,109 @@ int pcm_avail_update(struct pcm *pcm)
...
+			return 1;
+		}
+		int cnt = 0;
+		while(cnt < pcm->buffer_cnt - 1) {
```
Comment (chanijjani) [inline framework/src/tinyalsa/tinyalsa.c]: while (

File: framework/src/tinyalsa/tinyalsa.c [hunk 4]
```diff
@@ -1365,105 +1375,109 @@ int pcm_avail_update(struct pcm *pcm)
...
+				return -EPIPE;
+			} else {
+				break;
+			}
+		}
+		if (cnt == pcm->buffer_cnt -1) {
+			return 1;
+		}
+	
+		return oops(pcm, EINTR, "Recieved unexpected msg (id = %d) while waiting for deque message from kernel\n", msg.msgId);
```
Comment (JeongJunSik) [inline framework/src/tinyalsa/tinyalsa.c]: Received

**Pull request merged**